src/main/java/org/generationcp/middleware/domain/search_request/brapi/v2/SampleSearchRequestDTO.java

```diff
@@ -1,103 +1,135 @@
-        this.sampleDbId = sampleDbId;
-        this.observationUnitDbId = observationUnitDbId;
-        this.plateDbId = plateDbId;
-        this.germplasmDbId = germplasmDbId;
-        this.studyDbId = studyDbId;
-        this.externalReferenceID = externalReferenceID;
-        this.externalReferenceSource = externalReferenceSource;
-    }
-
-    public String getSampleDbId() {
-        return this.sampleDbId;
-    }
-
-    public void setSampleDbId(final String sampleDbId) {
-        this.sampleDbId = sampleDbId;
-    }
-
-    public String getObservationUnitDbId() {
-        return this.observationUnitDbId;
-    }
-
-    public void setObservationUnitDbId(final String observationUnitDbId) {
-        this.observationUnitDbId = observationUnitDbId;
-    }
-
-    public String getPlateDbId() {
-        return this.plateDbId;
-    }
-
-    public void setPlateDbId(final String plateDbId) {
-        this.plateDbId = plateDbId;
-    }
-
-    public String getGermplasmDbId() {
-        return this.germplasmDbId;
-    }
-
-    public void setGermplasmDbId(final String germplasmDbId) {
-        this.germplasmDbId = germplasmDbId;
-    }
-
-    public String getStudyDbId() {
-        return this.studyDbId;
-    }
-
-    public void setStudyDbId(final String studyDbId) {
-        this.studyDbId = studyDbId;
-    }
-
-    public String getExternalReferenceID() {
-        return this.externalReferenceID;
-    }
-
-    public void setExternalReferenceID(final String externalReferenceID) {
-        this.externalReferenceID = externalReferenceID;
-    }
-
-    public String getExternalReferenceSource() {
-        return this.externalReferenceSource;
-    }
-
-    public void setExternalReferenceSource(final String externalReferenceSource) {
-        this.externalReferenceSource = externalReferenceSource;
-    }
-
-    @Override
-    public int hashCode() {
-        return Pojomatic.hashCode(this);
-    }
-
-    @Override
-    public String toString() {
-        return Pojomatic.toString(this);
-    }
-
-    @Override
-    public boolean equals(final Object o) {
-        return Pojomatic.equals(this, o);
-    }
+public class SampleSearchRequestDTO extends SearchRequestDto {
+
+	private List<String> sampleDbIds;
+	private List<String> observationUnitDbIds;
+	private List<String> plateDbIds;
+	private List<String> germplasmDbIds;
+	private List<String> studyDbIds;
+	private List<String> externalReferenceIDs;
+	private List<String> externalReferenceSources;
+	private List<String> germplasmNames;
+	private List<String> studyNames;
+
+	public SampleSearchRequestDTO() {
+	}
+
+	public SampleSearchRequestDTO(final String sampleDbIds, final String observationUnitDbIds, final String plateDbIds,
+		final String germplasmDbIds, final String studyDbIds, final String externalReferenceIDs, final String externalReferenceSources) {
+		this.sampleDbIds = Collections.singletonList(sampleDbIds);
```

Comment:
perform StringUtils.isEmpty() before putting in singletonList. Also make the parameters singular